antlr-kotlin-runtime/src/commonTest/kotlin/MiniCalcParserTest.kt

```diff
@@ -53,6 +55,40 @@ class MiniCalcParserTest : BaseTest() {
         assertEquals("\n", newline.text)
     }
 
+    @test
+    fun testOperatorPrecedence() {
+        val input = CharStreams.fromString("1 - 2 + 3\n")
+
+        /*
+         * Test that the input is parsed like ((a - b) + c).
+         * The parse tree should look like:
+         *
+         *         plus
+         *        /   \
+         *     minus   \
+         *    /   \     \
+         *   1     2     3
+         */
+
+        val lexer = MiniCalcLexer(input)
+        val parser = MiniCalcParser(CommonTokenStream(lexer))
+
+        val plus = parser.expression() as MiniCalcParser.BinaryOperationContext
+        assertNotNull(plus.PLUS())
+
+        val minus = plus.left as MiniCalcParser.BinaryOperationContext
+        assertNotNull(minus.MINUS())
+
+        val one = minus.left as MiniCalcParser.IntLiteralContext
+        assertEquals(one.text, "1")
+
+        val two = minus.right as MiniCalcParser.IntLiteralContext
+        assertEquals(two.text, "2")
+
+        val three = plus.right as MiniCalcParser.IntLiteralContext
+        assertEquals(three.text, "3")
+    }
```

Comment:
@yaccoff I've changed the grammar back to the way it was and also changed to test to show the even more fundamental error.

Here we see that, _according to ANTLR_, `1 - 2 + 3` is supposed to be parsed as `(1 - 2) + 3`.
![image](https://user-images.githubusercontent.com/164758/158083491-8974a1af-d264-450b-9af6-b00d8ebedcea.png)

But `antlr-kotlin` generates code that parses `1 - 2 + 3` as `1 - (2 + 3)`. It does not adhere to the ANTLR standard!

Reply:
This bug basically makes `antlr-kotlin` unusable. Anyone can type `1 - 2 + 3` into a calculator (or [Google](https://www.google.com/search?q=1+-+2+%2B+3)) and get the answer: 2. But if I parse and evaluate that expression using `antlr-kotlin`, then I'll get -4!